src/OperationResponse.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace Google\GAX;
+
+/**
+ * Response object from a long running API method.
+ *
+ * The OperationResponse object is returned by API methods that perform
+ * a long running operation. It provides methods that can be used to
+ * poll the status of the operation, retrieve the results, and cancel
+ * the operation.
+ *
+ * To support a long running operation, the server must implement the
+ * Operations API, which is used by the OperationResponse object. If
+ * more control is required, it is possible to make calls against the
+ * Operations API directly instead of via the OperationResponse object
+ * using an OperationsClient instance.
+ */
+class OperationResponse
+{
+    const DEFAULT_POLLING_INTERVAL = 1.0;
+
+    private $operationName;
+    private $operationsClient;
+    private $operationReturnType;
+    private $metadataReturnType;
+    private $lastProtoResponse;
+    private $deleted = false;
+
+    /**
+     * OperationResponse constructor.
+     *
+     * @param string $operationName
+     * @param \Google\Longrunning\OperationsClient $operationsClient
+     * @param array $options {
+     *                       Optional. Options for configuring the Operation response object.
+     *
+     *     @type string $operationReturnType The return type of the longrunning operation.
+     *     @type string $metadataReturnType The type of the metadata returned in the Operation response.
+     *     @type \google\longrunning\Operation $lastProtoResponse A response already received from the server.
+     * }
+     */
+    public function __construct($operationName, $operationsClient, $options = [])
+    {
+        $this->operationName = $operationName;
+        $this->operationsClient = $operationsClient;
+        if (isset($options['operationReturnType'])) {
+            $this->operationReturnType = $options['operationReturnType'];
+        }
+        if (isset($options['metadataReturnType'])) {
+            $this->metadataReturnType = $options['metadataReturnType'];
+        }
+        if (isset($options['lastProtoResponse'])) {
+            $this->lastProtoResponse = $options['lastProtoResponse'];
+        }
+    }
+
+    /**
+     * Check whether the operation has completed.
+     *
+     * @return bool
+     */
+    public function isDone()
+    {
+         return (is_null($this->lastProtoResponse) || is_null($this->lastProtoResponse->getDone()))
+             ? false
+             : $this->lastProtoResponse->getDone();
+    }
+
+    /**
+     * Get the formatted name of the operation
+     *
+     * @return string The formatted name of the operation
+     */
+    public function getName()
+    {
+        return $this->operationName;
+    }
+
+    /**
+     * Poll the server in a loop until the operation is complete.
+     *
+     * Return true if the operation completed successfully, otherwise return
+     * false.
+     *
+     * The $pollSettings optional argument can be used to control the polling loop.
```

Comment:
This seems a little redundant since the description for the param is just below, but if we do keep it should this be `The $options optional argument`?